src/main/java/ArrayQueue.java

```diff
@@ -35,15 +35,20 @@ public void enqueue(T element) {
     /**
      * Doubles the size of the queue array and copy the
      * contents over.
+     *
+     * Mind where we copy from the original queue and
+     * the updates to front and rear
      */
     @SuppressWarnings("unchecked")
     private void expandCapacity() {
-        T[] newStack = (T[]) new Object[queue.length * 2];
+        T[] newQueue = (T[]) new Object[queue.length * 2];
         for (int i = 0; i < queue.length; ++i) {
-            newStack[i] = queue[i];
+            newQueue[i] = queue[front];
+            front = (front + 1) % queue.length;
         }
-        queue = newStack;
-
+        front = 0;
+        rear = size;
+        queue = newQueue;
```

Comment:
You the fixed the bug here I think, but your tests never actually exposed it. I suggest reverting this and writing a new test that should pass, bu actually fails. Then, once you have exposed the bug, reapply these changes and see if that causes the test to pass (while still passing all the other tests, of course).